x/ccv/provider/keeper/proposal.go

```diff
@@ -277,21 +281,36 @@ func (k Keeper) MakeConsumerGenesis(
 		bondedValidators = append(bondedValidators, val)
 	}
 
-	if prop.Top_N > 0 {
+	if topN, found := k.GetTopN(ctx, chainID); found && topN > 0 {
 		// in a Top-N chain, we automatically opt in all validators that belong to the top N
 		minPower := k.ComputeMinPowerToOptIn(ctx, chainID, bondedValidators, prop.Top_N)
 		k.OptInTopNValidators(ctx, chainID, bondedValidators, minPower)
 	}
 
-	nextValidators := k.ComputeNextEpochConsumerValSet(ctx, chainID, bondedValidators,
-		func(validator stakingtypes.Validator) bool {
-			return k.ShouldConsiderOnlyOptIn(ctx, chainID, validator)
+	nextValidators := k.FilterValidators(ctx, chainID, bondedValidators,
+		func(providerAddr types.ProviderConsAddress) bool {
+			// only consider opted-in validators
```

Comment:
Could you use FilterOptedInAndAllowAndDenylistedPredicate instead?

Reply:
Just fixed this.